### PR TITLE
icons, log grid name, component spawning on removeBlock

### PR DESCRIPTION
**Pull request by zrisher**

### CODE REVIEW

File: Data/Scripts/GardenConquest/Blocks/GridEnforcer.cs [hunk 1]
```diff
@@ -251,7 +255,7 @@ public enum VIOLATION_TYPE {
 		/// <param name="added"></param>
 		private void blockAdded(IMySlimBlock added) {
 			m_BlockCount++;
-			log("Block added to grid.  Count now: " + m_BlockCount, "blockAdded");
+			log("Block added to grid '" + m_Grid.DisplayName + "'. Count now: " + m_BlockCount, "blockAdded");
```
Comment (stackcollision) [inline Data/Scripts/GardenConquest/Blocks/GridEnforcer.cs:258]: DisplayName is being finnicky.  Leave this out.

File: Data/Scripts/GardenConquest/Blocks/GridEnforcer.cs [hunk 2]
```diff
@@ -416,11 +419,20 @@ public enum VIOLATION_TYPE {
...
+			// WAITING: once Keen accepts this PR
+			// https://github.com/KeenSoftwareHouse/SpaceEngineers/pull/52
+			// the below will spawn materials in the inventory if it exists and has space,
+			// or otherwise as floating objects in space,
+			//b.FullyDismount(inventory);
+			//b.MoveItemsFromConstructionStockpile(inventory);
+			//b.SpawnConstructionStockpile();
 
-			m_Grid.RemoveBlock(b);
+			b.CubeGrid.RemoveBlock(b);
```
Comment (stackcollision) [inline Data/Scripts/GardenConquest/Blocks/GridEnforcer.cs:435]: Previous line and your change are identical.  If a grid splits a new one is created and new block objects are allocated.  b.CubeGrid will still be m_Grid even if it is actually no longer attached.

Comment (zrisher) [inline Data/Scripts/GardenConquest/Blocks/GridEnforcer.cs:435]: Ah, whoops will remove, remnant of copy/pasting the logic from my SE core work.

File: Data/Scripts/GardenConquest/Blocks/GridEnforcer.cs (hunk 1 above)
Comment (zrisher) [inline Data/Scripts/GardenConquest/Blocks/GridEnforcer.cs:258]: Per discussion I will leave this in

**Comment by zrisher** [comment]
Closing in favor of #41

**Pull request closed without merging**